src/audio/mux/mux.c

```diff
@@ -262,6 +314,12 @@ static int mux_params(struct comp_dev *dev,
 
 	comp_info(dev, "mux_params()");
 
+	err = mux_update_buffer_channels_by_lookup_tables(dev, params->channels);
```

Comment:
To keep the existing cases unaffected, maybe adding a flag in config would be better to apply buffer channels by lookup tables and keep it off as default.
Or a config flag to indicate whether to propagate channels for stream parameter: 0 -> NOT propagate (the default behavior); 1 -> propagate. This is even simpler. Thoughts?

Reply:
@johnylin76 we were looking at something like this to propagate and align configuration within pipelines. https://github.com/thesofproject/sof/issues/1280
It's partially upstream atm, would this fit in with your case ?

Reply:
Thanks for the information.
As I see it #1280 had derived the solution that performs hw_params walk from DAI and then PCM params walk from host to overwrite, with comp_params() for supporting parameter tuning. It's like a turning point between PCM and HW params, for example:
```
host (16kHz) -----> SRC -----> DAI (48kHz)
```
SRC is the turning point, and sample rate is changed from PCM to HW preference. However it only works for cases with single turning point. If there are two turning points:
```
host (16kHz) -----> SRC1 -----(32kHz)----> SRC2 -----> DAI (48kHz)
```
How do we handle the parameter of the intermediate slice (from SRC1 to SRC2) if our expectation is neither PCM nor HW preference?

My case is kind of alike as two-turning-points case, where both DEMUX and MUX make channels update.
```
host (2-ch) ----> DEMUX ---(?-ch)---> MUX ----> DAI (4-ch)
```
IMHO it may be not enough to provide two choices (PCM or HW preference) only for the propagation of pipeline params.